WrathCombo/Combos/PvE/BRD/BRD.cs

```diff
@@ -654,9 +662,9 @@ protected override uint Invoke(uint actionID, uint lastComboMove, float comboTim
                             if (ActionReady(EmpyrealArrow) && JustUsed(WanderersMinuet)) // Used to ensure Empyreal arrow goes off as soon as possible in opener
                                 return EmpyrealArrow;
 
-                            if (canWeave)
+                            if (canWeave || !hasTarget)
                             {
-                                if (songNone)
+                                if (songNone && InCombat())
```

Comment:
If I'm in combat and un-target the boss, this looks to me like it would let me clip my GCDs if I have auto-target-on-attack turned on.

I would probably change this to be `if (canWeave && InCombat())`, for one, but since I want it to go whether I have a target or not ... this check might be superfluous, difficult to say from just what I see here.

Reply:
Similar issues at :432, :1184, :1219, at least.

Reply:
That's not a potential problem. Thats literally the point. 

It is so that your song can run when there is no boss to target. with no boss there is no gcd. with no gcd no weaving. no way to automatically run the songs. 

It is coded so that ONLY when in combat, and not targeting something, will the song work on a gcd. once targetting something, it is back on weave only.

Reply:
I do understand the point of this, I'm just saying there's more ways to lose a target than just actual downtime: I can arbitrarily un-target at any time, and if it's at song time I would clip.
I was saying that your target checks don't need to be *against* weave checks, it just adds the opportunity to clip, you get the same downtime improvements if you move the target check to happen after and separately from the weave check.

Reply:
If the target check is after the weave check. The it would still need to weave. That breaks the function I'm trying to give it. Running songs without something to target. The function of this NEEDS to be on the gcd not weaving when no target is available. 

There are plenty of bosses that step away for a time frame. Waiting on their adds to spawn or doing their super attack. Cagnazzo, zodiark, nidhogg to name a few. This is an actual issue for bard as the song timing is crucial to keep regardless of what the fight is doing. 

I know you are just trying to find the holes in it to make it better. The tiny risk of a gcd clip in the exact moment of a song switchover and someone randomly untargets the boss is far better than songs being out of rotation, which would then delay buff windows and throw off the whole fight for the bard and therefore the party.

Reply:
> The function of this NEEDS to be on the gcd not weaving when no target is available.

I see now, yes, I misunderstood the implicit logic there.

> The tiny risk of a gcd clip in the exact moment of a song switchover and someone randomly untargets the boss

Fair point.